src/lib/data/github.ts

```diff
@@ -24,33 +25,142 @@ function extractRepoInfo(repositoryUrl: string): { owner: string; repo: string }
 export class GitHubCollector {
   // Only store data that could be shared across multiple signals
   private starCount?: number;
+  private contributorsLastMonth?: number;
+  private repoInfo?: { owner: string; repo: string };
 
   async fetchPackage(repositoryUrl: string): Promise<void> {
     const repoInfo = extractRepoInfo(repositoryUrl);
     if (!repoInfo) {
       throw new Error(`Could not parse GitHub URL: ${repositoryUrl}`);
     }
 
+    this.repoInfo = repoInfo;
+
     try {
+      // Fetch basic repo data
       const response = await fetch(`https://api.github.com/repos/${repoInfo.owner}/${repoInfo.repo}`);
       if (!response.ok) {
         throw new Error(`GitHub API returned ${response.status}`);
       }
 
       const fullResponse = await response.json() as any;
       this.starCount = fullResponse.stargazers_count || 0;
+
+      // Fetch contributors from the past month
+      this.contributorsLastMonth = await this.fetchContributorsLastMonth();
     } catch (error) {
       throw new Error(`GitHub fetch failed: ${error}`);
     }
   }
 
+  private async fetchContributorsLastMonth(): Promise<number> {
+    if (!this.repoInfo) {
+      return 0;
+    }
+
+    try {
+      // Calculate date one month ago
+      const oneMonthAgo = new Date();
+      oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1);
+      const since = oneMonthAgo.toISOString();
+
+      // Fetch commits from the past month
+      const commitsUrl = `https://api.github.com/repos/${this.repoInfo.owner}/${this.repoInfo.repo}/commits?since=${since}&per_page=100`;
```

Comment:
this is the "fetch" part of the function. the rest of the function falls under "processing". based on what we talked about today, this should be split up along those lines too. and then see if you can further optimize the "fetch" methods, since they probably are calling similar APIs across metrics.